src/WebJobs.Script/Host/FunctionAppValidationService.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Diagnostics.Extensions;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.Host
+{
+    public sealed class FunctionAppValidationService : BackgroundService
+    {
+        private readonly IEnvironment _environment;
+        private readonly ILogger<FunctionAppValidationService> _logger;
+        private readonly IOptions<ScriptJobHostOptions> _scriptOptions;
+
+        public FunctionAppValidationService(
+            ILogger<FunctionAppValidationService> logger,
+            IOptions<ScriptJobHostOptions> scriptOptions,
+            IEnvironment environment)
+        {
+            _scriptOptions = scriptOptions ?? throw new ArgumentNullException(nameof(scriptOptions));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _environment = environment ?? throw new ArgumentNullException(nameof(environment));
+        }
+
+        protected override async Task ExecuteAsync(CancellationToken cancellationToken)
+        {
+            // Adding a delay to ensure that this validation does not impact the cold start performance
+            Utility.ExecuteAfterColdStartDelay(_environment, Validate, cancellationToken);
+
+            await Task.CompletedTask;
+        }
+
+        internal void Validate()
+        {
+            if (!_environment.IsPlaceholderModeEnabled() &&
+                !_scriptOptions.Value.IsDefaultHostConfig &&
+                !_scriptOptions.Value.IsStandbyConfiguration &&
+                Utility.IsDotnetIsolatedApp(environment: _environment) &&
+                !Directory.Exists(Path.Combine(_scriptOptions.Value.RootScriptPath, ScriptConstants.AzureFunctionsSystemDirectoryName)))
```

Comment:
Should there be any binding between this and some other version info, config, or legacy indication that the isolated worker model is in use? If an app has been deployed with a much older version of the isolated worker model, is it possible that they might still run afoul of this even though they have a valid deployed payload?

Reply:
Do you mean that there were older versions of the isolated model that did not require the `.azurefunctions` folder to work correctly? I'm not aware of this scenario. Do you have any additional information that can help me repro the scenario and then add appropriate checks for it.